File: packages/core/src/privacy/PrivacyPolicy.ts
import { RedactionType } from "./Redaction";

export interface PrivacyPolicyConfig {
  enabledRedactions: RedactionType[];
  logRedactions: boolean;
  strictMode: boolean;
}

export const defaultPrivacyPolicy: PrivacyPolicyConfig = {
  enabledRedactions: ["email", "phone", "token", "api_key", "credit_card", "ssn"],
  logRedactions: true,
  strictMode: true,
};

File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/privacy/PrivacyPolicy.kt
package ai.mobileagentos.compose.privacy

data class PrivacyPolicyConfig(
    val enabledRedactions: List<RedactionType>,
    val logRedactions: Boolean
)

val defaultPrivacyPolicy = PrivacyPolicyConfig(
    enabledRedactions = listOf(
        RedactionType.EMAIL,
        RedactionType.PHONE,
        RedactionType.TOKEN,
        RedactionType.API_KEY,
        RedactionType.CREDIT_CARD,
        RedactionType.SSN
    ),
    logRedactions = true
)